File: front-end-tests/selenium-web-driver-tests/IdeaCenterPOM/IdeaCenterPOM/Pages/CreateIdeaPage.cs
using OpenQA.Selenium;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace IdeaCenterPOM.Pages
{
	public class CreateIdeaPage : BasePage
	{
		public CreateIdeaPage(IWebDriver driver) : base(driver) { }

		public override string PageUrl => base.PageUrl + "/Ideas/Create";
		public IWebElement TitleField => _driver.FindElement(By.XPath("//input[@name='Title']")); // 2 - 70
		public IWebElement ImgUrlField => _driver.FindElement(By.XPath("//input[@name='Url']"));
		public IWebElement DescriptionField => _driver.FindElement(By.XPath("//textarea[@name='Description']")); // 3 - 400
		public IWebElement CreateButton => _driver.FindElement(By.XPath("//button[@type='submit']"));
		public IWebElement MainErrorMsg => _driver.FindElement(By.XPath("//div[@class='text-danger validation-summary-errors']//li"));
		public IWebElement TitleErrorMsg => _driver.FindElement(By.XPath("//span[@data-valmsg-for='Title']"));
		public IWebElement DescErrorMsg => _driver.FindElement(By.XPath("//span[@data-valmsg-for='Description']"));

		public void CreateIdea(string title, string imgUrl, string desc)
		{
			TitleField.Clear();
			TitleField.SendKeys(title);

			ImgUrlField.Clear();
			ImgUrlField.SendKeys(imgUrl);

			DescriptionField.Clear();
			DescriptionField.SendKeys(desc);

			CreateButton.Click();
		}
	}
}

File: front-end-tests/playwright-tests/pages/CreateIdeaPage.js
export class CreateIdeaPage {
    constructor(page) {
        this.page = page;
        this.titleInput = page.getByRole('input', { name: 'Title' });
        this.pictureUrlInput = page.getByRole('input', { name: 'Url' });
        this.descriptionInput = page.getByRole('textarea', { name: 'Description' });
        this.createBtn = page.getByRole('button', { type: 'submit' });
    }

    async goto() {
        await this.page.goto('http://softuni-qa-loadbalancer-2137572849.eu-north-1.elb.amazonaws.com:83/Ideas/Create');
    }

    async fillIdeaInfo({ title, pictureUrl, description }) {
        if (title) {
            await this.titleInput.fill(title);
        }

        if (pictureUrl) {
            await this.pictureUrlInput.fill(pictureUrl);
        }

        if (description) {
            await this.descriptionInput.fill(description);
        }
    }

    async clickCreateBtn() {
        await this.createBtn.click();
    }
}